packages/react-native-sdk/src/hooks/useScreenShareButton.ts

```diff
@@ -0,0 +1,125 @@
+   *
+   */
+  onScreenShareStoppedHandler?: () => void,
+  /**
+   * Handler to be called when the permissions to stream screen share media is missing
+   *
+   */
+  onMissingScreenShareStreamPermission?: () => void
+) => {
+  const call = useCall();
+  const { useLocalParticipant, useCallSettings, useOwnCapabilities } =
+    useCallStateHooks();
+  const callSettings = useCallSettings();
+  const ownCapabilities = useOwnCapabilities();
+  const hasScreenSharingPermissions = ownCapabilities?.includes(
+    OwnCapability.SCREENSHARE
+  );
+  const isScreenSharingEnabledInCall = callSettings?.screensharing.enabled;
+
+  const onScreenShareStartedHandlerRef = useRef(onScreenShareStartedHandler);
+  onScreenShareStartedHandlerRef.current = onScreenShareStartedHandler;
+  const onScreenShareStoppedHandlerRef = useRef(onScreenShareStoppedHandler);
+  onScreenShareStoppedHandlerRef.current = onScreenShareStoppedHandler;
+
+  const iosScreenShareStartedFromSystem = useIsIosScreenshareBroadcastStarted();
+  const prevIosScreenShareStartedFromSystem = usePrevious(
+    iosScreenShareStartedFromSystem
+  );
+
+  const localParticipant = useLocalParticipant();
+  const hasPublishedScreenShare =
+    localParticipant && hasScreenShare(localParticipant);
+
+  // listens to iOS screen share broadcast started event from the system
+  useEffect(() => {
+    const run = async () => {
+      if (Platform.OS !== 'ios') {
+        return;
+      }
+      if (
+        iosScreenShareStartedFromSystem &&
+        !prevIosScreenShareStartedFromSystem
+      ) {
+        onScreenShareStartedHandlerRef.current?.();
+        call?.screenShare.enable();
+      } else if (
+        !iosScreenShareStartedFromSystem &&
+        prevIosScreenShareStartedFromSystem
+      ) {
+        onScreenShareStoppedHandlerRef.current?.();
+        call?.screenShare.disable(true);
+      }
+    };
+    run();
+  }, [
+    call,
+    iosScreenShareStartedFromSystem,
+    prevIosScreenShareStartedFromSystem,
+  ]);
+
+  const onPress = async () => {
+    if (!hasScreenSharingPermissions) {
+      const logger = getLogger(['useScreenShareButton']);
+      logger(
+        'info',
+        'User does not have permissions to stream the screen share media, calling onMissingScreenShareStreamPermission handler if present'
+      );
+      onMissingScreenShareStreamPermission?.();
+    }
+    if (!hasPublishedScreenShare) {
+      if (Platform.OS === 'ios') {
+        const reactTag = findNodeHandle(screenCapturePickerViewiOSRef.current);
+        await NativeModules.ScreenCapturePickerViewManager.show(reactTag);
+        // After this the iOS screen share broadcast started/stopped event will be triggered
+        // and the useEffect listener will handle the rest
+      } else {
+        try {
+          await call?.screenShare.enable();
+          onScreenShareStartedHandler?.();
+        } catch (e) {
+          // ignored.. user didnt allow the screen share in the popup
```

Comment:
Please don't hide exceptions. It makes debugging harder than necessary. Can we have at least a log statement here?